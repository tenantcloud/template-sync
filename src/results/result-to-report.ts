import { SyncResult } from "../sync";

export function syncResultToReport(result: SyncResult): string {
	const repositories = result.repositories.map(
		({ url, branch }) => `- <${url}> at branch \`${branch}\``,
	);

	return `## The following repositories were used for sync:

${repositories.join("\n")}`;
}
