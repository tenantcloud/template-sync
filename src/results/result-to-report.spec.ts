import { syncResultToReport } from "./result-to-report";

describe("syncResultToReport", () => {
	it("converts result to markdown string", () => {
		const report = syncResultToReport({
			repositories: [
				{ url: "https://github.com/first/repo", branch: "alpha" },
				{ url: "https://github.com/second/repo", branch: "master" },
			],
		});

		expect(report).toMatchSnapshot();
	});
});
