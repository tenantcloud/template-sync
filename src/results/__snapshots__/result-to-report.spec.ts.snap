// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`syncResultToReport converts result to markdown string 1`] = `
"## The following repositories were used for sync:

- <https://github.com/first/repo> at branch \`alpha\`
- <https://github.com/second/repo> at branch \`master\`"
`;
